pulser-core/pulser/backend/config.py

```diff
@@ -14,28 +14,223 @@
+            times will use this configuration's 'default_evaluation_times'.
+        default_evaluation_times: The default times at which observables
+            are computed. Can be a sequence of relative times between 0
+            (the start of the sequence) and 1 (the end of the sequence).
+            Can also be specified as "Full", in which case every step in the
+            emulation will also be an evaluation times.
+        initial_state: The initial state from which emulation starts. If
+            specified, the state type needs to be compatible with the emulator
+            backend. If left undefined, defaults to starting with all qudits
+            in the ground state.
+        with_modulation: Whether to emulate the sequence with the programmed
+            input or the expected output.
+        interaction_matrix: An optional interaction matrix to replace the
+            interaction terms in the Hamiltonian. For an N-qudit system,
+            must be an NxN symmetric matrix where entry (i, j) dictates
+            the interaction coefficient between qudits i and j, ie it replaces
+            the C_n/r_{ij}^n term.
+        prefer_device_noise_model: If the sequence's device has a default noise
+            model, this option signals the backend to prefer it over the noise
+            model given with this configuration.
+        noise_model: An optional noise model to emulate the sequence with.
+            Ignored if the sequence's device has default noise model and
+            `prefer_device_noise_model=True`.
     """
 
-    backend_options: dict[str, Any] = field(default_factory=dict)
+    observables: Sequence[Observable]
+    default_evaluation_times: np.ndarray | Literal["Full"]
+    initial_state: StateType | None
+    with_modulation: bool
+    interaction_matrix: pm.AbstractArray | None
+    prefer_device_noise_model: bool
+    noise_model: NoiseModel
+
+    def __init__(
+        self,
+        *,
+        observables: Sequence[Observable] = (),
+        # Default evaluation times for observables that don't specify one
+        default_evaluation_times: Sequence[SupportsFloat] | Literal["Full"] = (
+            1.0,
+        ),
+        initial_state: StateType | None = None,  # Default is ggg...
+        with_modulation: bool = False,
+        interaction_matrix: ArrayLike | None = None,
+        prefer_device_noise_model: bool = False,
+        noise_model: NoiseModel = NoiseModel(),
+        **backend_options: Any,
+    ) -> None:
+        """Initializes the EmulationConfig."""
+        for obs in observables:
+            if not isinstance(obs, Observable):
+                raise TypeError(
+                    "All entries in 'observables' must be instances of "
+                    f"Observable. Instead, got instance of type {type(obs)}."
+                )
+
+        if default_evaluation_times != "Full":
+            eval_times_arr = np.array(default_evaluation_times, dtype=float)
```

Comment:
why is this a `np.array` but below you use `AbstractArray`?

Reply:
You mean for `interaction_matrix`? I use `AbstractArray` because I expect people might want to differentiate against it